lib/utils/apiRequests.ts

```diff
@@ -0,0 +1,34 @@
+import { CustomerDTO, EmployeeDTO, ProjectDTO, TaskDTO, TimesheetDTO } from '../types/dto';
+import { get, post, put, ResponseWithStatus } from './fetch';
+
+const projectEndpointURL = `${process.env.NEXT_PUBLIC_API_URL}/project`;
+const customerEndpointURL = `${process.env.NEXT_PUBLIC_API_URL}/customer`;
+const taskEndpointURL = `${process.env.NEXT_PUBLIC_API_URL}/task`;
+const employeeEndpointURL = `${process.env.NEXT_PUBLIC_API_URL}/employee`;
+const timesheetEndpointURL = `${process.env.NEXT_PUBLIC_API_URL}/timesheet`;
+const projectIdEndpointURL = (id: number): string => `${process.env.NEXT_PUBLIC_API_URL}/project/${id}`;
+
+export const getProject = (projectId: number): Promise<ResponseWithStatus<ProjectDTO>> =>
+    get<ProjectDTO>(projectIdEndpointURL(projectId));
+export const listProjects = (): Promise<ResponseWithStatus<ProjectDTO[]>> => get<ProjectDTO[]>(projectEndpointURL);
+
+export const listCustomers = (): Promise<ResponseWithStatus<CustomerDTO[]>> => get<CustomerDTO[]>(customerEndpointURL);
```

Comment:
These type signatures suck: I'd like to define
```
export type ApiRequest<T> = Promise<ResponseWithStatus<T>>
```
but this doesn't work due to [a bug](https://github.com/microsoft/TypeScript/issues/27987) in typescript itself.